src/main/java/com/microsoft/graph/content/MSBatchResponseContent.java

```diff
@@ -188,8 +204,20 @@ public String nextLink() {
         if (batchResponse == null)
             return null;
         try {
-            final Map<String, Request> batchRequestsHashMap = new LinkedHashMap<>();
             final JsonObject requestJSONObject = requestBodyToJSONObject(batchResponse.request());
+            final String baseUrl = batchResponse.request().url().toString().replace("$batch", "");
+            return createBatchRequestsHashMap(baseUrl, requestJSONObject);
+        } catch (IOException ex) {
+            ex.printStackTrace();
+            return null;
+        }
+    }
+    private Map<String, Request> createBatchRequestsHashMap(@Nonnull final String baseUrl, @Nonnull final JsonObject requestJSONObject) {
+        if(baseUrl == null || baseUrl == "" || requestJSONObject == null) {
```

Comment:
nit: you could use UrlValidator to provide more comprehensive validation of the baseUrl. I'm not sure how a dev could mess up the URL at this point

Reply:
The URL parsing is done down below the stack. Plus I believe [UrlValidator is in an apache dependency](https://commons.apache.org/proper/commons-validator/apidocs/org/apache/commons/validator/routines/UrlValidator.html) we don't have at this moment. I could do the URL parsing (using the URL ctor), but it's already done below and I'd have to check for null before doing do anyway.